api/src/services/extensions.ts

```diff
@@ -86,7 +86,44 @@ export class ExtensionsService {
 		}
 
 		if ('meta' in data && 'enabled' in data.meta) {
-			await this.knex('directus_extensions').update({ enabled: data.meta.enabled }).where({ name: key });
+			if (bundle !== null) {
+				const parent = await this.readOne(null, bundle);
+
+				if (parent.schema?.type !== 'bundle') {
+					await this.knex('directus_extensions').update({ enabled: data.meta.enabled }).where({ name: key });
+				} else {
+					if (parent.schema.partial === false) {
+						// if a bundle is non-partial do not allow toggling a child
+						throw new InvalidPayloadError({ reason: 'Unable to disable an entry for a bundle marked as non partial' });
+					}
+
+					await this.knex('directus_extensions').update({ enabled: data.meta.enabled }).where({ name: key });
+
+					const child = await this.knex('directus_extensions')
+						.where('name', 'LIKE', `${bundle}/%`)
+						.where({ enabled: true })
+						.first();
+
+					if (!child && parent.meta.enabled) {
+						// if all children are disabled then ensure parent bundle is disabled
+						await this.knex('directus_extensions').update({ enabled: false }).where({ name: parent.name });
+					} else if (child && !parent.meta.enabled) {
+						// if at least one child is enabled then ensure parent bundle is enabled
+						await this.knex('directus_extensions').update({ enabled: true }).where({ name: parent.name });
+					}
+				}
+			} else {
+				await this.knex('directus_extensions').update({ enabled: data.meta.enabled }).where({ name: key });
+
+				const extension = await this.readOne(bundle, name);
+
+				if (extension.schema?.type === 'bundle') {
+					// if a bundle is toggled, then toggle all of its children
+					await this.knex('directus_extensions')
+						.update({ enabled: data.meta.enabled })
+						.where('name', 'LIKE', `${name}/%`);
+				}
+			}
```

Comment:
can we reduce the amount of nested if statements here by adding some early returns? Something like:
```suggestion
			if (bundle === null) {
				await this.knex('directus_extensions').update({ enabled: data.meta.enabled }).where({ name: key });

				const extension = await this.readOne(bundle, name);

				if (extension.schema?.type === 'bundle') {
					// if a bundle is toggled, then toggle all of its children
					await this.knex('directus_extensions')
						.update({ enabled: data.meta.enabled })
						.where('name', 'LIKE', `${name}/%`);
				}
				
				this.extensionsManager.reload();
				return;
			}

			const parent = await this.readOne(null, bundle);

			if (parent.schema?.type !== 'bundle') {
				await this.knex('directus_extensions').update({ enabled: data.meta.enabled }).where({ name: key });
				
				this.extensionsManager.reload();
				return;
			}
			
			if (parent.schema.partial === false) {
				// if a bundle is non-partial do not allow toggling a child
				throw new InvalidPayloadError({ reason: 'Unable to disable an entry for a bundle marked as non partial' });
			}

			await this.knex('directus_extensions').update({ enabled: data.meta.enabled }).where({ name: key });

			const child = await this.knex('directus_extensions')
				.where('name', 'LIKE', `${bundle}/%`)
				.where({ enabled: true })
				.first();

			if (!child && parent.meta.enabled) {
				// if all children are disabled then ensure parent bundle is disabled
				await this.knex('directus_extensions').update({ enabled: false }).where({ name: parent.name });
			} else if (child && !parent.meta.enabled) {
				// if at least one child is enabled then ensure parent bundle is enabled
				await this.knex('directus_extensions').update({ enabled: true }).where({ name: parent.name });
			}
```

Reply:
Done!